pkg/sysinfo/sysinfo_linux_test.go

```diff
@@ -56,3 +59,36 @@ func TestCgroupEnabled(t *testing.T) {
 		t.Fatal("cgroupEnabled should be true")
 	}
 }
+
+func TestNew(t *testing.T) {
+	sysInfo := New(false)
+	require.NotNil(t, sysInfo)
+
+	sysInfo = New(true)
+	require.NotNil(t, sysInfo)
+
+	// Check if AppArmor is supported. then it must be TRUE , else FALSE
+	if _, err := os.Stat("/sys/kernel/security/apparmor"); !os.IsNotExist(err) {
```

Comment:
I have refactored the code. Please let me know if any more changes are required.